### ISSUE TITLE
"Hide DS4 Controller" is broken on Windows 10

### ISSUE THREAD

**Issue by thomasboyt** [Jays2Kings/DS4Windows#26]
On Windows 10 build 10074, it seems like DS4Windows is no longer able to hide the DS4 controller. I've tried every combination of reinstalling the drivers, reinstalling DS4Windows, and restarting my computer possible, but it just doesn't seem to function.

This is using the DS4 controller plugged in to the PC, not through Bluetooth. Curiously, the controller shows up as "Wireless Controller" in the Device Manager, but that may be normal. Attempting to disable the Wireless Controller device also disables the 360 Controller device, so I'm not sure there's a way on my end to fix this.

**Comment by thomasboyt** [comment]
Ah, it looks like this may be a duplicate of https://github.com/Jays2Kings/DS4Windows/issues/6 - has there been any progress on a potential fix?

**Comment by thomasboyt** [comment]
[Here's an Input Mapper forum thread](https://inputmapper.com/forum/4/390) that discusses the same issue, the users here seem to have seen the same behavior I've seen in DS4Windows - I also saw exclusive mode randomly start/stop working over the past few days since I've installed Windows 10.

**Comment by Jays2Kings** [comment]
Build 10074 sometimes frees the controller, however like Build 9879 it's the system grabbing the controller, and there is no fix from my end, only Microsoft's.

**Comment by Dainesis** [comment]
Hello, I'm a user of Windows 10 RTM ( build 10240 ) and here is how I could make the ds 4 work in all my games : 
- Launch DS4Windows in admin mode
- Uncheck Hide ds4 if checked
- Check Hide ds4
- Launch the game
- While in game, unplug the ds4 then plug it
- Replug the ds4 BUT spam buttons right before you do it, I guess it's to set the priority to the 360

Some steps could be unnecessary, but this is how I managed to make it work with all my games. ( tested on dark souls 1 & 2, phantasy star online 2 and project cars )

**Comment by nefarius** [comment]
@Jays2Kings out of curiosity I'm currently debugging The Witcher 3 which has "native" DS4 support and I'm not able to figure out how the game engine interacts with the DS4:

_dinput8.dll_
- Doesn't use `IDirectInputDevice8::GetDeviceData`
- Doesn't use `IDirectInputDevice8::GetDeviceInfo`
- Doesn't use `IDirectInputDevice8::GetDeviceState`

_HID.dll_
- Doesn't use `HidD_GetInputReport`
- Doesn't use `HidP_GetData`
- Doesn't use `HidD_GetPreparsedData`

Any ideas?

**EDIT:** never mind, I over-thought this; they access the `\\?\hid#vid_054c&pid_05c4#...` device path through `CreateFile` and `ReadFile` to receive the input reports :)

**Comment by nefarius** [comment]
[Success!](https://www.youtube.com/watch?v=ktMC-shuvRI)

**Comment by miller-time** [comment]
@nefarius will that work with DS4Windows in general? I want to get exclusive access to my DS4, but I get:

> Warning: Could not open DS4 exclusively. You must quit other applications like Steam, Uplay before activating the 'Hide DS4 Controller' option.

Clearly something is hanging onto it. I don't have anything running and I tried the trick where you kill explorer.exe even. Is it possible to use your HideDS4.dll to prevent the controller from being used at all (so that I can enable the exclusive mode)?

**Comment by ghost** [comment]
Here is what works for me:
- Open the device manager.
- Go to Human Interface Devices.
- If you use USB, disable and enable HID-compliant game controller.
- If you use BT, disable and enable Bluetooth-HID-device.

Don't restart when asked!
Hide DS4 works after that until I reconnect my controller.

**Comment by WilliamIPark** [comment]
@syne- You legend! That worked for me perfectly.

**Comment by adamelders** [comment]
@syne- That worked for me as well. Just a note - the controller needs to be plugged in, otherwise the "HID-compliant game controller" won't show up. Thanks!

**Comment by miller-time** [comment]
@syne- this worked!!

I needed to:
- quit Steam & Uplay
- stop DS4Windows
- disable/re-enable HID-compliant game controller
- start DS4Windows

and for the first time ever, exclusive mode works. THANKS!!

Controller now works perfectly in Rocket League and The Crew (those are the only 2 games I play w/ controller at the moment)

**Comment by nefarius** [comment]
@miller-time nope, the DLL has to be loaded into every process which tries to open the DS4.

**Comment by WilliamIPark** [comment]
I'm back again just to ask: can this fix be implemented into the program itself?

**Comment by teokp** [comment]
not sure if its the same issue but you may want to try my build here
https://github.com/teokp/DS4Windows/releases
it automatically re-enables the device when exclusive mode fails

**Comment by Jays2Kings** [comment]
I've implemented teokp's fix for this in the newest build of DS4Windows, Thanks so much for your help, a tremendous fix.

**Comment by teokp** [comment]
@Jays2Kings Its great to see my commits merged into the new version, however, there is a small issue with the fix.

I recently discovered that SetupDiChangeState is recommended to only be called by the class installer of the device so I switched to use SetupDiCallClassInstaller instead. The problem is that SetupDiCallClassInstaller is platform specific for some reason, meaning that if you are using 64-bit Windows, DS4Windows must also be built in 64-bit for it to work.

**Comment by Jays2Kings** [comment]
Ah, I see. The thing is I don't want to split DS4Windows into multiple versions, and in non debug mode, the DS4 can at connect exclusively to usb and bluetooth regardless of CPU configuration, however if I see a true issue with this, I will split up DS4Windows I guess.

**Comment by RPDragonforce98** [comment]
@syne- when i do this my left analog stick is not recognized and wont do anything. i know it works perfectly

**Comment by Kaypain** [comment]
Im very confusing none of this works for me. When i'm plugged the controller doesn't work correctly on Blue tooth it does. The Hide DS4 doesn't work as well. I don't understand the source foler, what am I supposed to do with it?

**Comment by Dave0611** [comment]
I tried Disabling and Enabling Bluetooth HID but do i need to do that everytime i restart my computer? i tried to restart my computer and it cant run the exclusive mode again. thanks

**Comment by Mboxer511** [comment]
@syne- Your solution worked for me too!

> Here is what works for me:
> 
>     Open the device manager.
>     Go to Human Interface Devices.
>     If you use USB, disable and enable HID-compliant game controller.
>     If you use BT, disable and enable Bluetooth-HID-device.
> 
> Don't restart when asked!
> Hide DS4 works after that until I reconnect my controller.

**Comment by xperia64** [comment]
I recently had this stop working for me as well, but I found a workaround.
Nvidia's services seem to be hooking the controller even after I exit Steam, so to enable Exclusive Mode reliably, I quit Steam, then stop these services in Task Manager:
NvContainerLocalSystem
NvContainerNetworkService
NVDisplay.ContainerLocalService
NvTelemetryContainer

After restarting DS4Windows, the controller then works in Exclusive mode as expected.

**Comment by BroseidonMOO** [comment]
@xperia64 The issue is specifically the in-game overlay. If you turn it off, stop/start DS4, it'll regain exclusive mode until the controller is disconnected again. You don't need to kill all of the Nvidia processes.

**Comment by UnstoppableBeef** [comment]
@xperia64 After hours and hours of trying to get Exclusive Mode to work today I came across your post and it instantly fixed my issue. I always wondered if there was some process somewhere causing the problem, and I even suspected it might be an Nvidia program, but I was lost in the endless list of processes running on my machine. Thanks for the tip!

**Comment by MilosMoravac** [comment]
Guys, disabling the IN-GAME overlays worked for me. Thanks everybody for helping!!

**Comment by DaveKap** [comment]
Came to say that the nvidia in-game overlay was a problem for me as well. I wish the DS4Windows guys could somehow get around that :(

**Comment by oarfan24** [comment]
been several years, but i still have a problem where my emulator (retroarch) recognizes a controller both as a DS4 and a xinput device. When I follow the directions here (disable/renable bt HID) retroarch only recognizes 1 input correctly, but the controls don't work. any advice?

**Comment by ghost** [comment]
@oarfan24 I don’t know if it will solve the issue but checkout the fork by Ryochan7 which is still actively developed. Maybe it’s fixed in their version already or if not open an issue there.

**Comment by oarfan24** [comment]
thanks for the quick response - yeah that is the one i have - very
frustrating. i think i'll return the controllers and go with something else
at this point.

**Comment by AliNej2323** [comment]
@xperia64 what is the solution for people without nvidia graphic card (Radeion rx for me)? i just want to play fifa on my pc but origin doesnt let me. i can connect my controller but the hide ds4 controller option doesnt work exclusively.

**Comment by mika-n** [comment]
#654 
This Jay2Kings/DS4Windows site is obsolete and out-dated. You should take a look at the Ryochan7/DS4Windows instead for more recent version.
